hordes-ui-mod.js

```diff
@@ -382,941 +392,1071 @@
+                            $element.parentNode.removeChild($element);
+                        });
+                    });
+                });
+                // And the close button for our custom UI
+                document.querySelector('.js-close-custom-settings').addEventListener('click', () => {
+                    const $customSettingsWindow = document.querySelector('.js-custom-settings');
+                    $customSettingsWindow.parentNode.removeChild($customSettingsWindow);
+                });
+            });
+        },
+
+        // This creates the initial chat context menu DOM (which starts as hidden)
+        function createChatContextMenu() {
+            if (document.querySelector('.js-chat-context-menu')) {
+                return;
+            }
+
+            let contextMenuHTML = `
     			<div class="js-name">...</div>
 				<div class="choice" name="party">Party invite</div>
 				<div class="choice" name="whisper">Whisper</div>
 				<div class="choice" name="friend">Friend</div>
 				<div class="choice" name="unfriend">Unfriend</div>
 				<div class="choice" name="block">Block</div>
 			`
-    		document.body.appendChild(makeElement({
-    			element: 'div',
-    			class: 'panel context border grey js-chat-context-menu',
-    			content: contextMenuHTML,
-    		}));
-
-    		const $chatContextMenu = document.querySelector('.js-chat-context-menu');
-    		$chatContextMenu.querySelector('[name="party"]').addEventListener('click', () => {
-				modHelpers.partyPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="whisper"]').addEventListener('click', () => {
-    			modHelpers.whisperPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="friend"]').addEventListener('click', () => {
-    			modHelpers.friendPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="unfriend"]').addEventListener('click', () => {
-    			modHelpers.unfriendPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="block"]').addEventListener('click', () => {
-    			modHelpers.blockPlayer(tempState.chatName);
-    		});
-    	},
-
-    	// This opens a context menu when you click a user's name in chat
-    	function chatContextMenu() {
-    		const addContextMenu = ($name, name) => {
-    			$name.classList.add('js-is-context-menu-initd');
-    			// Add name to element so we can target it in CSS when filtering chat for block list
-    			$name.setAttribute('data-chat-name', name);
-
-    			const showContextMenu = clickEvent => {
-    				// TODO: Is there a way to pass the name to showChatContextMenumethod, instead of storing in tempState?
-    				tempState.chatName = name;
-    				modHelpers.showChatContextMenu(name, {x: clickEvent.pageX, y: clickEvent.pageY});
-    			};
-    			$name.addEventL
```

Comment:
Can you remove all the svelte classes here? How painful would that be?

Trying really hard to avoid leveraging svelte classes for anything. They're a bit more unstable, more likely to change with updates. If you need styles from them, feel free to create `uidom-x` classes for custom styles, and copy the svelte CSS into there.

Reply:
these are used for styling to apply the common UI theme. they are just the common classes used for the panels

Reply:
these are used by the main css bundle for the common theme UI.